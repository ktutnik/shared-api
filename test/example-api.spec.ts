

import supertest from "supertest"
import createApp from "../src/app"
import { closeConnection, createApplication, createUser, ignoreProps } from "./helper"

afterEach(async () => {
    await closeConnection()
})

describe("Example API", () => {
    it("Should able to access basic plan with basic application", async () => {
        const plum = await createApp({ mode: "production" })
        const { app } = await createApplication(plum)
        const { body } = await supertest(plum.callback())
            .get("/api/example/app")
            .set("x-api-key", app.apiKey)
            .expect(200)
        expect(body).toMatchSnapshot({ ...ignoreProps, apiKey: expect.any(String) })
    })
    it("Should able to access premium plan with premium application", async () => {
        const plum = await createApp({ mode: "production" })
        const { app } = await createApplication(plum, "Premium")
        const { body } = await supertest(plum.callback())
            .get("/api/example/premium")
            .set("x-api-key", app.apiKey)
            .expect(200)
        expect(body).toMatchSnapshot()
    })
    it("Should not able to access premium plan with basic application", async () => {
        const plum = await createApp({ mode: "production" })
        const { app } = await createApplication(plum, "Basic")
        await supertest(plum.callback())
            .get("/api/example/premium")
            .set("x-api-key", app.apiKey)
            .expect(403)
    })
})