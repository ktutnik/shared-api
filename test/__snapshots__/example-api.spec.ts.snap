// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example API Should able to access basic plan with basic application 1`] = `
Object {
  "active": true,
  "apiKey": Any<String>,
  "createdAt": Any<String>,
  "deleted": false,
  "id": Any<Number>,
  "name": "Lorem Ipsum",
  "type": "Basic",
  "updatedAt": Any<String>,
}
`;

exports[`Example API Should able to access premium plan with premium application 1`] = `
Object {
  "message": "Premium",
}
`;
